### PR TITLE
Allow `Date` styles to be hidden in log output

### LINKED ISSUE #44: Unable to disable `Date` group in log output?
Hi!

I'm struggling to disable the highlight group for `Date`, i.e., hide the `Date` column in log output

From README:
```md
To disable a highlight group, either comment it out or delete it.
```

I've attempted:
```toml
# 1) comment out
#[groups.date]

# 2) set disabled = true
[groups.date]
disabled = true
```

It looks like I can set disabled on `Date` and it should work? I may be missing something obvious here 😅 
https://github.com/bensadeh/tailspin/blob/504e47fbc7a325f3ebacb10346a449bba2a64f26/src/theme/mod.rs#L62-L68

### PR DESCRIPTION
**Pull request by peterkos**
## Description

This PR allows for a Date style to be hidden in log output. Fixes #44!

```toml
# config.toml
[groups.date]
date = { hidden = true}
time = { hidden = true}
zone = { hidden = true}
```

Example:
```log
# Before
2022-09-09 11:44:54,508 [Worker-2: Loading available Gradle versions] INFO  # other stuff
2022-09-09 11:47:47,193 [Worker-3: Importing Maven projects] INFO           # other stuff
2022-09-09 11:47:47,783 [Worker-3: Importing Maven projects] INFO           # other stuff

# After
[Worker-2: Loading available Gradle versions] INFO  # other stuff
[Worker-3: Importing Maven projects] INFO           # other stuff
[Worker-3: Importing Maven projects] INFO           # other stuff
```

## Contents

- `Style` now has a `hidden: Bool` flag
- Refactored `DateHighlighter.apply(_:)` logic
- Added tests
- Added [indoc](https://lib.rs/crates/indoc) crate for nice test macro formatting

### CODE REVIEW

**Comment by peterkos** [comment]
Only remaining issue I've found is that it likes to print an extra space after all fields are disabled 😵 

I think this is a result of the regex parsing in `apply`, will try to provide more info from the last time I looked at that bug

File: src/highlighters/date.rs [hunk 1]
```diff
@@ -6,17 +6,25 @@ use crate::theme::Style;
 use crate::types::Highlight;
 
 pub struct DateHighlighter {
-    date: String,
-    time: String,
-    zone: String,
+    date_ansi: Option<String>,
+    time_ansi: Option<String>,
+    zone_ansi: Option<String>,
```
Comment (peterkos) [inline src/highlighters/date.rs:11]: I opted for `Option<String>` as we can have a `hidden` style (field not displayed), and `disabled` style (field not styled)

**Comment by bensadeh** [comment]
Hi @peterkos !

Thank you very much for taking the time to create this PR for hiding date segments. I am happy to add this feature to `tailspin`.

I've been thinking about the date highlighter for a while, and I would like to split the date highlighter into a "date only" highlighter and "time only" highlighter. I think it would help to simplify both the regex output (like the space before `INFO`) as well as the logic for hiding the segments.

Since this will be affecting this PR, I'll open a new PR with the date and time highlighter split, as well as the hiding of the segment. Will be happy to get your feedback once it is up.

**Comment by bensadeh** [comment]
Implemented in `1.6.0`.

**Pull request closed without merging**